XSLT/tsudctomods.xsl

```diff
@@ -0,0 +1,154 @@
+        <mods xmlns="http://www.loc.gov/mods/v3" version="3.5"
+            xmlns:xlink="http://www.w3.org/1999/xlink"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://www.loc.gov/mods/v3 http://www.loc.gov/standards/mods/v3/mods-3-5.xsd">
+            
+            <!-- title -->
+            <xsl:apply-templates select="dc:title"/>
+            
+            <!-- identifier -->
+            <xsl:apply-templates select="dc:identifier"/>
+            
+            <!-- abstract or thumbnail -->
+            <xsl:apply-templates select="dc:description"/>
+            
+            <!-- creator -->
+            <xsl:apply-templates select="dc:creator"/>
+            
+            <!-- originInfo> -->
+            <xsl:apply-templates select="dc:date"/>
+            
+            <!-- subject(s) -->
+            <xsl:apply-templates select="dc:subject"/>
+            
+            <!-- relatedItem -->
+            <xsl:apply-templates select="dc:source"/>
+            
+            <!-- publisher or recordContentSource -->
+            <recordInfo><recordContentSource>Tennessee State University</recordContentSource></recordInfo>
+            
+            <!-- rights or accessCondition -->
+            <accessCondition type="use and reproduction" xlink:href="http://rightsstatements.org/vocab/CNE/1.0/">Copyright Not Evaluated</accessCondition>
+            
+        </mods>
+    </xsl:template>
+    
+    <!-- title -->
+    <xsl:template match="dc:title">
+        <titleInfo><title><xsl:apply-templates/></title></titleInfo>
+    </xsl:template>
+    
+    <!-- identifier / object URL / typeOfResource based off collection -->
+    <xsl:template match='dc:identifier'>
+        <xsl:choose>
+            <xsl:when test="matches(., '[0-9]$')">
+                <location>
+                    <url usage="primary" access="object in context"><xsl:apply-templates/></url>
+                </location>
+            </xsl:when>
+            <xsl:when test="contains(., 'library-digital-collections/')">
+                <typeOfResource>still image</typeOfResource>
+            </xsl:when>
+            <xsl:when test="contains(., 'catalogues')">
+                <typeOfResource>text</typeOfResource>
+            </xsl:when>
+            <xsl:when test="contains(., 'yearbooks')">
+                <typeOfResource>text</typeOfResource>
+            </xsl:when>
+        </xsl:choose>
+    </xsl:template>
+    
+    <!-- abstract or thumbnail -->
+    <xsl:template match="dc:description">
+        <xsl:choose>
+            <xsl:when test="starts-with(., 'https://')">
+                <location>
+                    <url access="preview"><xsl:apply-templates/></url>
+                </location>
+            </xsl:when>
+            <xsl:otherwise>
+                <abstract><xsl:apply-templates/></abstract>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
```

Comment:
The way this is written creates 2 location nodes, but there should only be one.

Reply:
Do you have any suggestions since they've sadly put their thumbnail links in dc:description? I was unsure how to handle this since the two location/url s span across different matches.

Reply:
Perhaps I can try transforming the existing identifier URL instead of taking the straight thumbnail URL from the description field.